### ISSUE TITLE
conv2d crashes when compute gradient

### ISSUE THREAD

**Issue by VictoriaGriffith** [tensorflow/tensorflow#57401]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.9

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.9

### Current Behaviour?

```shell
Session crashes when computing gradient for `tf.nn.conv2d`. For the inputs in the example below, `tf.nn.conv2d` can generate a correct (empty) output tensor with shape `(1, 16, 0, 1)`. However, when I compute gradient with GradientTape, the program crashes with error message (core dumped).
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
filters = tf.random.uniform([3, 3, 1, 1], dtype=tf.float32)
strides = 2
padding = "VALID"

input = tf.random.uniform([1, 34, 1, 1], dtype=tf.float32)

with tf.GradientTape(persistent=True,) as g:
  g.watch(input)
  res_backward = tf.nn.conv2d(input, filters, strides=strides, padding=padding)
print(res_backward)
grad = g.gradient(res_backward, input) # this will crash
```

### Relevant log output

```shell
tf.Tensor([], shape=(1, 16, 0, 1), dtype=float32)
core dumped
```
</details>

**Comment by sushreebarsa** [comment]
@VictoriaGriffith I tried to replicate the issue on colab and it crashes on gpu runtime. Could you please find the gists of [cpu](https://colab.research.google.com/gist/sushreebarsa/585c03e413417554c92e9c027e1407d7/57401.ipynb#scrollTo=JI7pV-XkTMeX) and [gpu](https://colab.research.google.com/gist/sushreebarsa/72f3daa0eb3785a6f4538f8bffcd2a52/57401.ipynb#scrollTo=0RitoivmOi6y) runtime here? Please confirm the issue ?
Thank you!

**Comment by VictoriaGriffith** [comment]
@sushreebarsa yes, thanks for replicating this issue!

**Comment by sachinprasadhs** [comment]
This could be due to the different check implementation for both CPU and GPU, could you please refer the comment [here](https://github.com/tensorflow/tensorflow/issues/57174#issuecomment-1246288003).
